### ISSUE TITLE
not all code paths return a value PHP(PHP0405), when void is specified as a @return

### ISSUE THREAD

**Issue by addshore** [DEVSENSE/phptools-docs#73]
I saw this problem reported:

![image](https://user-images.githubusercontent.com/3308769/90020761-3335c600-dca8-11ea-81c7-4f160c68ba32.png)

Looking at the abstract function being implemented I see that the `@return` annotation specifies `bool`, so the problem seems okay at this stage, as the implementation should probably return a bool in all cases (rather than void).

Rather than specify a return at the bottom of all implemented methods I went ahead to add `void` to the `@return` annotation:

![image](https://user-images.githubusercontent.com/3308769/90020943-7859f800-dca8-11ea-9948-feefdccf3822.png)

After doing this the same "problem" was reported.

When specifying `void` as a possible return value of a method
I would not expect to see `PHP0405` as it shouldn't apply?

Minimum reproduction:

```
<?php

abstract class A {

	/**
	 * @return bool|void
	 */
	abstract public function method();
}

class B extends A {

	public function method() {
		if ( 1 == 2) {
			return true;
		}
		// No other return value
	}

}
```

**Comment by jakubmisek** [comment]
I see, `PHP0405` is reported whenever there is an inconsistent `return`.

We should check for PHPDoc whether this inconsistency is desired.

BTW in result, `bool|void` actually means `bool|NULL` because there is no Void in PHP :)

**Comment by addshore** [comment]
I'd argue that `bool|void` and `bool|null` are technically slightly different.

https://wiki.php.net/rfc/void_return_type
https://www.php.net/manual/en/language.pseudo-types.php

> void as a return type means that the return value is useless.

void doesn't mean null, and I guess when specifying a return type as `bool|void` you could return anything really, but `bool|null` you would only expect bools or nulls.

> We should check for PHPDoc whether this inconsistency is desired.
Do the PHPxxxx problem numbers match up to some php / phpdoc spec that I have never read before? :)

**Comment by jakubmisek** [comment]
You're right, for the editor it means the value is useless. In run-time, the value is NULL. 

Those PHPxxxx numbers are actually pretty old (>12 years), we can prepare some kind of table with compatibility with other sniff tools!

**Comment by jakubmisek** [comment]
ok, so this only happens when there is missing the PHPDoc, phptools won't use the PHPDoc from abstract method

**Comment by jakubmisek** [comment]
thanks, fixed since 1.05069
